### PR TITLE
Add DeBERTa

### PR DESCRIPTION
**Pull request by jimypbr**
Waiting on next poplar SDK to land to main
- [x] Get base to compile and fit on IPU
- [x] Get base to train on SQuAD

### CODE REVIEW

File: optimum/graphcore/trainer.py [hunk 1]
```diff
@@ -1208,14 +1204,11 @@ def _load_state_dict_in_model(self, state_dict):
...
 
-        # TODO: check if this is needed.
-        # if self.training_model and self.training_model.isAttachedToDevice():
-        #     self.training_model.copyWeightsToDevice()
-
-        # if self.inference_model and self.inference_model.isAttachedToDevice():
-        #     self.inference_model.copyWeightsToDevice()
         if len(load_result.missing_keys) != 0:
-            logger.warn(f"There were missing keys in the checkpoint model loaded: {load_result.missing_keys}.")
+            if self.model._keys_to_ignore_on_save is not None and set(load_result.missing_keys) != set(
```
Comment (michaelbenayoun) [inline optimum/graphcore/trainer.py:1208]: @jimypbr I saw that you removed those tests when I rebased, should I keep those? 
What was the rational when removing them?

Comment (jimypbr) [reply optimum/graphcore/trainer.py:1210]: I think this is right to put them back. The check got removed when I was fixing the roberta checkpointing bug where there was an extra `tie_weights` that was breaking things.

**Review by michaelbenayoun — COMMENTED** [review]
Why?
I think the changes we've done were good no?
The idea is to simplify things.

**Comment by jimypbr** [comment]
> Why? I think the changes we've done were good no? The idea is to simplify things.

These changes were breaking the checkpointing with BERT. It was causing an error after it dumps the first checkpoint.

**Comment by michaelbenayoun** [comment]
Ok, we will work on that in later PRs then.

**Pull request merged**